fix: use dl.k8s.io, not kubernetes-release bucket

rjsadow opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
There are still references to https://storage.googleapis.com/kubernetes-release instead of https://dl.k8s.io/

dl.k8s.io is the correct advertised download host and will eventually move to be fastly shielding a fully community-owned bucket

ref: https://github.com/kubernetes/k8s.io/issues/2396
ref: https://github.com/kubernetes/kubernetes/issues/117949

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

rjsadow commented:
/release-note-none

rjsadow commented:
/remove-kind deprecation
/release-note-none

rjsadow commented:
/release-note-none

BenTheElder commented on CHANGELOG/CHANGELOG-1.2.md:
```diff
@@ -58,27 +58,27 @@
 filename | sha256 hash
 -------- | -----------
-[kubernetes.tar.gz](https://storage.googleapis.com/kubernetes-release/release/v1.2.7/kubernetes.tar.gz) | `53db157923c17fa7a0addb3e4dfe7d1b9194b9266a87d371a251d5bb790a1832`
-[kubernetes-src.tar.gz](https://storage.googleapis.com/kubernetes-release/release/v1.2.7/kubernetes-src.tar.gz) | `e6e46831706743d8263581d0575507cf5ffc265096d22e5e84cf1c3ae925db5e`
```
We should change these super old changelogs, however this does prove that we've been advertising with dl.k8s.io since 1.10 or so.

I think the plan for now is to leave kubernetes-release available but convert it to read-only once we can migrate to using a bucket under a community account. So these can stay, but we want to remove _current_ references.

rjsadow replied:
I've reverted the CHANGELOG updates.

BenTheElder reviewed (commented):
Please revert the CHANGELOG changes, otherwise this looks good. Thanks!

BenTheElder commented:
/approve

BenTheElder commented:
/priority important-soon
/triage accepted

rjsadow committed:
fix: use dl.k8s.io, not kubernetes-release bucket

This PR updates changes related references to the legacy
release bucket, excluding CHANGELOG updates.

Signed-off-by: Ricky Sadowski <richard.j.sadowski@gmail.com>

rjsadow commented:
/retest

BenTheElder commented:
/lgtm
